test/integration/referral.spec.ts

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { ReferralError } from '../../src/ports/referral/types'
+import { test } from '../components'
+
+test('Referral', ({ components }) => {
+  const endpoint = 'https://referral-server.decentraland.zone/referral-progress'
+  let apiKey: string
+  let invitedUserAddress: string
+  let mockFetch: jest.Mock
+
+  beforeEach(async () => {
+    jest.clearAllMocks()
+
+    mockFetch = jest.fn()
+    global.fetch = mockFetch
+
+    apiKey = await components.config.requireString('REFERRAL_SERVER_API_KEY')
+    invitedUserAddress = '0x36359cd12e64c150a347f1dd3ff95bf68b46b33f'
+  })
+
+  describe('When updateReferral is called', () => {
+    describe('with valid body', () => {
```

Comment:
As we're going to be building a context where the API successfully responds to our request, it should be described as:
```suggestion
  describe('when succesfully updating the referral', () => {
```